airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/unstructured_parser.py

```diff
@@ -0,0 +1,123 @@
+
+            if filetype not in self._supported_file_types():
+                raise RecordParseError(FileBasedSourceError.ERROR_PARSING_RECORD, filename=file.uri)
+
+            return {
+                "content": {"type": "string"},
+                "document_key": {"type": "string"},
+            }
+
+    def parse_records(
+        self,
+        config: FileBasedStreamConfig,
+        file: RemoteFile,
+        stream_reader: AbstractFileBasedStreamReader,
+        logger: logging.Logger,
+        discovered_schema: Optional[Mapping[str, SchemaType]],
+    ) -> Iterable[Dict[str, Any]]:
+        with stream_reader.open_file(file, self.file_read_mode, None, logger) as file_handle:
+            markdown = self._read_file(file_handle, file.uri)
+            yield {
+                "content": markdown,
+                "document_key": file.uri,
+            }
+
+    def _read_file(self, file_handle: IOBase, file_name: str) -> str:
+        from unstructured.file_utils.filetype import FileType
+        from unstructured.partition.auto import partition
+        from unstructured.partition.md import optional_decode
+
+        filetype = self._get_filetype(file_handle, file_name)
+
+        if filetype == FileType.MD:
+            file_content: bytes = file_handle.read()
+            decoded_content: str = optional_decode(file_content)
+            return decoded_content
+        if filetype not in self._supported_file_types():
+            raise RecordParseError(FileBasedSourceError.ERROR_PARSING_RECORD, filename=file_name)
+
+        file: Any = file_handle
+        if filetype == FileType.PDF:
+            # for PDF, read the file into a BytesIO object because some code paths in pdf parsing are doing an instance check on the file object and don't work with file-like objects
+            file_handle.seek(0)
+            file = BytesIO(file_handle.read())
+            file_handle.seek(0)
+
+        elements = partition(file=file, metadata_filename=file_name)
+        return self._render_markdown(elements)
+
+    def _get_filetype(self, file: IOBase, file_name: str) -> Any:
+        from unstructured.file_utils.filetype import detect_filetype
+
+        # set name to none, otherwise unstructured will try to get the modified date from the local file system
+        if hasattr(file, "name"):
+            file.name = None
+
+        return detect_filetype(
+            file=file,
+            file_filename=file_name,
+        )
+
+    def _supported_file_types(self) -> List[Any]:
+        from unstructured.file_utils.filetype import FileType
+
+        return [FileType.MD, FileType.PDF, FileType.DOCX]
+
+    def _render_markdown(self, elements: List[Any]) -> str:
+        return "\n\n".join((self._convert_to_markdown(el) for el in elements))
+
+    def _convert_to_markdown(self, el: Any) -> str:
+        from unstructured.documents.elements import Formula, ListItem, Title
+
+        if type(el) == Title:
```

Comment:
Generally `isinstance()` is preferred over `==` for type checking...


```suggestion
        if isinstance(el, Title):
```